server/controllers/chatController.js

```diff
@@ -0,0 +1,221 @@
+                profile: "$_id",
+                picture: "$profilePicture",
+              },
+            },
+          },
+        ]);
+        //Bring all the data together and create array of objects
+        let conversations = profilesData.map(item => {
+          return {
+            firstName: item._id.firstName,
+            lastName: item._id.lastName,
+            picture: item._id.picture,
+            profileId: item._id.profile,
+            userId: userProfiles.filter(element => {
+              return (
+                element._id.profileId.toString() == item._id.profile.toString()
+              );
+            })[0]._id.userId,
+          };
+        });
+        conversations.map(item => {
+          item.lastMessage = lastMessagesList.filter(element => {
+            return (
+              element.companionUserId.toString() === item.userId.toString()
+            );
+          })[0].lastMessage;
+          item.chatId = lastMessagesList.filter(element => {
+            return (
+              element.companionUserId.toString() === item.userId.toString()
+            );
+          })[0].chatId;
+        });
+        //sort chats the way latest message will be on top, and chats with no messages will be at the bottom
+        conversations.sort((a, b) => {
+          if (a.lastMessage) {
+            if (b.lastMessage) {
+              if (a.lastMessage.timeCreated > b.lastMessage.timeCreated) {
+                return -1;
+              } else {
+                return 1;
+              }
+            } else {
+              if (a.lastMessage.timeCreated > 0) {
+                return -1;
+              }
+            }
+          } else if (!a.lastMessage && !b.lastMessage) {
+            if (a.chatId > b.chatId) {
+              return -1;
+            }
+          }
+        });
+        res.json({ error: false, data: conversations });
+      }
+    );
+  } catch (err) {
+    next(createError(500, err.message));
+  }
+};
+
+exports.sendMessage = async (req, res, next) => {
+  try {
+    const { content, chatId } = req.body;
+    const sender = req.user.id;
+    await Chat.findById(chatId, async (err, chat) => {
+      if (err) console.log(err);
+      messageId = chat.messages;
+      Message.findById(messageId, (err, messages) => {
+        if (err) console.log(err);
+        messages.messages.push({
+          sender: sender,
+          content: content,
+          timeCreated: moment().format(),
+          wasRead: false,
+          chatId: chat._id,
+        });
+        messages.save();
+        res.json({ error: false, message: "Message was delivered" });
+      });
+    });
+  } catch (err) {
+    next(createError(500, err.message));
+  }
+};
+
+exports.historyOfMessages = async (req, res, next) => {
+  try {
+    const { chatId } = req.body;
+    await Chat.findById(chatId, async (err, chat) => {
+      if (err) console.log(err);
+      await Message.findById(chat.messages, (err, messages) => {
```

Comment:
You can use `.populate("messages")` on the first find to save yourself the extra code